src/core/ddsi/src/ddsi_deadline.c

```diff
@@ -78,7 +78,9 @@ extern inline void ddsi_deadline_reregister_instance_locked (struct ddsi_deadlin
 void ddsi_deadline_register_instance_real (struct ddsi_deadline_adm *deadline_adm, struct deadline_elem *elem, ddsrt_mtime_t tprev, ddsrt_mtime_t tnow)
 {
   ddsrt_circlist_append(&deadline_adm->list, &elem->e);
+  elem->deadlines_missed = 0;
```

Comment:
I know it is nitpicking, but I would reorder them and assign `t_last_trigger` first, then base `t_deadline` on that.